src/kimmdy/kmc.py

```diff
@@ -34,19 +49,27 @@ class KMCResult:
     time_start
         Time, from which the reaction starts. The reaction changes the
         geometry/topology of this timestep and continues from there. [ps]
+    time_start_index
+        Index into the list of timesteps or rates of the timestep where the reaction starts
     """
 
-    recipe: Recipe = field(default_factory=lambda: Recipe([], [], []))
-    reaction_probability: Optional[list[float]] = None
-    time_delta: Optional[float] = None
-    time_start: Optional[float] = None
+    recipe: Recipe
+    reaction_probability: list[float] | None
+    time_delta: float
+    time_start: float
+    time_start_index: int
+    time_start_index_within_plugin: int | None = None
+
+
+@dataclass
+class KMCRejection:
```

Comment:
i think the name gets confused with rejection-free/rejection KMC. How about calling KMCResult KMCAccept and KMCRejection KMCNull

Reply:
On closer look, only one of those `KMCRejection` results represents an actual rejection, the others represent something going wrong in the selection, like and empty list of of rates, so this would be the third type, which can just be regular `None`.

Reply:
gave it its own type for clarity